packages/core/src/types.ts

```diff
@@ -274,6 +284,11 @@ export type UpdateAccountCenterAction = {
   payload: AccountCenter | Partial<AccountCenter>
 }
 
+export type UpdateConnectModalAction = {
+  type: 'update_connect_modal'
+  payload: ConnectModalOptions | Partial<ConnectModalOptions>
```

Comment:
```suggestion
  payload: Partial<ConnectModalOptions>
```

I know we have this pattern elsewhere, but I am pretty sure that we can simplify to just using the `Partial` as that just makes all parameters optional in the object, but would still allow for the full type to be passed there.